### ISSUE TITLE
Rule: Enforce ordering of keys in test cases

### ISSUE THREAD

**Issue by not-an-aardvark** [not-an-aardvark/eslint-plugin-eslint-plugin#1]
Requires the properties of a test case to be placed in the given order.

```js
/* eslint eslint-plugin/test-case-property-ordering: ["error",
  ["code", "output", "options", "parserOptions", "errors"]
] */

// invalid; wrong order
{
  code: "foo",
  options: ["baz"],
  output: "bar"
}

// valid; correct order
{
  code: "foo",
  output: "bar",
  options: ["baz"]
}

// invalid; extra properties should need to be placed afterwards, I think
{
  code: "foo",
  env: { es6: true }
  output: "bar",
  options: ["baz"],
}

// valid
{
  code: "foo",
  output: "bar",
  options: ["baz"],
  env: { es6: true }
}
```
